src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -2,35 +2,33 @@
 
 namespace Microsoft.CST.OpenSource.PackageManagers
 {
-    using Microsoft.CST.RecursiveExtractor;
     using Microsoft.Extensions.Caching.Memory;
     using Microsoft.CST.OpenSource.Model;
     using System;
     using System.Collections.Generic;
     using System.IO;
     using System.Linq;
     using System.Net.Http;
-    using System.Text;
     using System.Text.Json;
     using System.Text.RegularExpressions;
     using System.Threading.Tasks;
     using Utilities;
     using Version = SemanticVersioning.Version;
     using PackageUrl;
 
-    public class BaseProjectManager
+    public abstract class BaseProjectManager
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="BaseProjectManager"/> class.
         /// </summary>
-        public BaseProjectManager(IHttpClientFactory httpClientFactory, string destinationDirectory)
+        public BaseProjectManager(IHttpClientFactory httpClientFactory, string destinationDirectory = ".")
```

Comment:
No, because only NuGet right now uses any implementation of `IManagerPackageActions`, so there isn't a need to have it as part of the base project manager as well I don't think.
I separated the `IHttpClientFactory` out of the `IManagerPackageActions` (previously `IManagerProvider`) as it wasn't necessary to do so and caused some confusing scenarios where every project manager had a `IManagerPackageActions` but only used it as an object containing an `IHttpClientFactory`, adding an unnecessary wrapper essentially.

Reply:
> No, because only NuGet right now uses any implementation of `IManagerPackageActions`, so there isn't a need to have it as part of the base project manager as well I don't think.

If the purpose of the refactor was to have an interface for provider actions but you only use it for Nuget its not an interface.  Its just class with a complex type.

>  I separated the `IHttpClientFactory` out of the `IManagerPackageActions` (previously `IManagerProvider`) as it wasn't necessary to do so and caused some confusing scenarios where every project manager had a `IManagerPackageActions` but only used it as an object containing an `IHttpClientFactory`, adding an unnecessary wrapper essentially.

I think this is far more confusing.

Reply:
I understand the confusion, I was struggling to decide how to handle it myself.
I personally think that the `IManagerPackageActions` shouldn't contain an `IHttpClientFactory`, at least not currently.
I think one thing that could be worth doing is a follow up PR that implements a new `IManagerPackageActions` class for each manager and then separate out the logic accordingly. I think then it would make sense to have the `IHttpClientFactory` as a part of `IManagerPackageActions` but we would also want it in the `ProjectManager` implementation as well I'd imagine.